src/Core/Commander/ManagerCommands.js

```diff
@@ -71,43 +76,44 @@ define('Core/Commander/ManagerCommands', [
             return this.commandsLength()===0;
         };
 
-        ManagerCommands.prototype.runAllCommands = function() {
-
-
-            if (this.commandsLength() === 0)
-            {
-                return Promise.resolve(0);
-            }
-
-            return Promise.all(this.arrayDeQueue(16))
-                .then(function () {
-                    // if (this.commandsLength() <= 16)
-                    this.scene.wait(1);
-                    // else
-                    //     this.scene.renderScene3D();
-                    return this.runAllCommands();
-                }.bind(this));
+        ManagerCommands.prototype.resetExecutedCommandsCount = function() {
+            this.counters.executedCommands = 0;
         };
 
-        ManagerCommands.prototype.arrayDeQueue = function(number) {
-
-            var nT = number === undefined ? this.queueAsync.length : number;
-
-            var arrayTasks = [];
-
-            while (this.queueAsync.length > 0 && arrayTasks.length < nT) {
-                var command = this.deQueue();
-
-                if(command) {
-                    var layer = command.layer;
-                    var provider = this.providers[layer.protocol];
-                    if (provider) {
-                        arrayTasks.push(provider.executeCommand(command));
+        ManagerCommands.prototype.executeCommands = function() {
+
+            var command;
+            var launchCommand = function(cmd) {
+                this.counters.runningCommands++;
+                var layer = cmd.layer;
+                var p = this.providers[layer.protocol];
+                if(p) {
+                    var promise = p.executeCommand(cmd);
+                    if(promise) {
+                        promise.then(function() {
+                            this.counters.runningCommands--;
+                            this.counters.executedCommands++;
+                        }.bind(this));
+                    } else {
+                        this.counters.runningCommands--;
+                        this.counters.executedCommands++;
```

Comment:
Still no per-host command limit ? (see my earlier comment on why it's important)